chat/src/main/java/com/github/twitch4j/chat/TwitchChat.java

```diff
@@ -314,18 +259,27 @@ public TwitchChat(EventManager eventManager, CredentialManager credentialManager
         this.removeChannelOnJoinFailure = removeChannelOnJoinFailure;
         this.maxJoinRetries = maxJoinRetries;
         this.chatJoinTimeout = chatJoinTimeout;
-        this.wsPingPeriod = wsPingPeriod;
 
-        // Create WebSocketFactory and apply proxy settings
-        this.webSocketFactory = new WebSocketFactory();
-        if (proxyConfig != null)
-            proxyConfig.applyWs(webSocketFactory.getProxySettings());
+        // init connection
+        if (websocketConnection == null) {
+            this.connection = new WebsocketConnection(spec -> {
+                spec.baseUrl(baseUrl);
```

Comment:
Should we allow `IExponentialStrategy` to be set in chat/pubsub builder so users can override details like `maxRetries` (and we'll pass it along via the spec)